### ISSUE TITLE
Code Deprecation and missing method calls in Chapter02/02_memory_overview/03_image_scaling

### ISSUE THREAD

**Issue by thatrobotguy** [PacktPublishing/Learn-CUDA-Programming#5]
This code example has 2 issues. In the example `Chapter02/02_memory_overview/03_image_scaling`, I tried compiling it with the standard `nvcc -o <exec name> <cudafile>.cu` syntax `nvcc -o runme image_scaling.cu`, but I get a deprecation error.
```
image_scaling.cu(90): warning: conversion from a string literal to "char *" is deprecated

image_scaling.cu(90): warning: conversion from a string literal to "char *" is deprecated

image_scaling.cu: In function ‘int main(int, char**)’:
image_scaling.cu:82:67: warning: ‘cudaError_t cudaThreadSynchronize()’ is deprecated [-Wdeprecated-declarations]
         returnValue = (cudaError_t)(returnValue | cudaThreadSynchronize());
                                                                   ^
/usr/local/cuda-10.0/bin/../targets/x86_64-linux/include/cuda_runtime_api.h:947:46: note: declared here
 extern __CUDA_DEPRECATED __host__ cudaError_t CUDARTAPI cudaThreadSynchronize(void);
                                              ^~~~~~~~~~~~~~~~~~~~~
/tmp/tmpxft_000010a4_00000000-10_image_scaling.o: In function `main':
tmpxft_000010a4_00000000-5_image_scaling.cudafe1.cpp:(.text+0x159): undefined reference to `get_PgmPpmParams(char*, int*, int*)'
tmpxft_000010a4_00000000-5_image_scaling.cudafe1.cpp:(.text+0x1ba): undefined reference to `scr_read_pgm(char*, unsigned char*, int, int)'
tmpxft_000010a4_00000000-5_image_scaling.cudafe1.cpp:(.text+0x535): undefined reference to `scr_write_pgm(char*, unsigned char*, int, int, char*)'
collect2: error: ld returned 1 exit status
```
There are actually 2 problems with this code. Code deprecation was only the first one.

I searched for a solution for the deprecation issue and I found one [here](https://stackoverflow.com/questions/13485018/cudastreamsynchronize-vs-cudadevicesynchronize-vs-cudathreadsynchronize). Basically, the command that should be used is to switch out `cudaThreadSynchronize` with `cudaDeviceSynchronize` on line 82 of `image_scaling.cu`.

The second issue has to do with the `pgm` file parsing method calls. For some reason, the main script cannot find the correct method calls by looking through the header file, so the second line in `image_scaling.cu` needs to change from `#include"scrImagePgmPpmPackage.h"` to `#include"scrImagePgmPpmPackage.cpp"`. Only then does it find the method definitions.

TL;DR
The code works after some edits, and I wanted to point out what those edits were.

I am not an expert on importing directives, but I hypothesize that the original `cpp` syntax is correct, but the `nvcc` compiler is looking for header/class files in the wrong order.

**Comment by packt-pradeeps** [comment]
The code files have been updated.
